packages/mobile/src/tokens/saga.ts

```diff
@@ -266,44 +269,58 @@ export async function getERC20TokenContract(tokenAddress: string) {
   return new kit.web3.eth.Contract(erc20.abi, tokenAddress)
 }
 
-export async function getERC20TokenBalance(token: StoredTokenBalance, address: string) {
-  let balance = null
-  try {
-    const contract = await getERC20TokenContract(token.address)
-    balance = await contract.methods.balanceOf(address).call()
-  } catch (error) {
-    Logger.error(TAG, `error fetching balance for ${token.name}`, error)
-  }
-  return balance
+interface BlockscoutTokenBalance {
+  balance: string
+  contractAddress: string
+  decimals: string
+  name: string
+  symbol: string
+  type: string
 }
 
-export function* fetchReadableTokenBalance(address: string, token: StoredTokenBalance) {
-  const balance: number | null = yield call(getERC20TokenBalance, token, address)
-  return {
-    ...token,
-    balance:
-      balance || balance === 0
-        ? new BigNumber(balance)
-            .dividedBy(new BigNumber(10).exponentiatedBy(token.decimals))
-            .toString()
-        : null,
+export function* fetchTokenBalancesFromBlockscout(address: string) {
```

Comment:
Also, it would be nice to test this function by really calling blockscout api. So we can be sure that the returned model is the one that we are expecting. Maybe we can use `alfajores` api, and use a fixed testing address?
Not sure if that is the best approach, but I think having a e2e test for this api call adds some value :)

Reply:
> Can we add that this returns a BlockscoutTokenBalance?

We can't actually :( These generator functions have typing issues, that's why you see errors with red underlines for all `yield` usages except if you put in the type manually, which isn't super safe of course. I think there was talk in the past to fix this, but we never ended up prioritizing it I think. cc @jeanregisser 

re: calling the real blockscout api I think that's where e2e tests come in. We usually mock these things in unit tests and have an e2e test that tests the entire flow, there's an action item pending which is adding e2e tests for this that will call the actual API and test everything, does that sound like enough to you?

Reply:
Yes we want to use `redux-typed-saga`.
By default vscode uses its own TypeScript version, but we can switch to our project version by clicking `{}` next to TypeScript in the status bar. This avoid seeing all the errors.